pawn/scripting/include/ripext/http.inc

```diff
@@ -157,6 +160,39 @@ methodmap HTTPClient < Handle
 	// @param value      Optional value to pass to the callback function.
 	public native void UploadFile(const char[] endpoint, const char[] path, HTTPFileCallback callback, any value = 0);
 
+	// URL encode the given string.
+	//
+	// Provided by curl https://curl.haxx.se/libcurl/c/curl_easy_escape.html
+	//
+	// @param dest		Destination to store the escaped string.
+	// @param maxlength Maximum length of the destination.
+	// @param source	A string need to escape.
+	public native void Escape(char[] dest, int maxlength, const char[] source);
+
+	public void GetWithParams(const char[] endpoint, StringMap params, HTTPRequestCallback callback, any value = 0)
+	{
+		char path[SM_RIPEXT_MAX_PATH_LENGTH];
+		// Encode params
+		StringMapSnapshot snapshot = params.Snapshot();
```

Comment:
You need to delete the snapshot handle after use.

Reply:
Yep 😆 My wrong